csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -2230,29 +2417,32 @@ private predicate pathIntoCallable(
   )
 }
 
-/** Holds if data may flow from a parameter given by `sc` to a return of kind `kind`. */
+/**
+ * Holds if data may flow from a parameter `p` with access path `apIn`, to a
+ * return of kind `kind` with access path `apOut`.
+ */
 pragma[nomagic]
 private predicate paramFlowsThrough(
-  ReturnKindExt kind, CallContextCall cc, SummaryCtx sc, AccessPath ap, Configuration config
+  ParameterNode p, ReturnKindExt kind, CallContextCall cc, AccessPath apIn, AccessPath apOut,
+  Configuration config
 ) {
   exists(PathNodeMid mid, ReturnNodeExt ret |
     mid.getNode() = ret and
     kind = ret.getKind() and
     cc = mid.getCallContext() and
-    sc = mid.getSummaryCtx() and
+    TSummaryCtxSome(p, apIn) = mid.getSummaryCtx() and
```

Comment:
Why unfold the summary context?  We might as well keep it wrapped up.

Reply:
Right, as long as the parameter type is `TSummaryCtxSome` and not `SummaryCtx`.